giskard_cicd/loaders/huggingface_inf_model.py

```diff
@@ -0,0 +1,49 @@
+from giskard import Model
+import pandas as pd
+import numpy as np
+from time import sleep
+
+def extract_scores(data):
+    scores = []
+    
+    if isinstance(data, dict):
+        # extract 'score' value if it exists
+        score = data.get('score')
+        if score is not None:
+            scores.append(score)
+        
+        # Recursively process dictionary values
+        for value in data.values():
+            scores.extend(extract_scores(value))
+    
+    elif isinstance(data, list):
+        # recursively process list elements
+        for element in data:
+            scores.extend(extract_scores(element))
+    
+    return scores
+
+def predict_from_inference(df: pd.DataFrame, query) -> np.ndarray:
```

Comment:
Can we add `predict_from_text_classification_inference` to implement this?

Then, in the `predict_from_inference`, it can accept the model type and dispatch to the corresponding function.

Reply:
Do you mean to distinguish text classification and other kinds of classification?

I would like to distinguish the model by different model wrappers instead of passing too many parameters. I don't think these parameters would settle in short term and it would cause fewer troubles to restructure it if we separate them completely for now.

Reply:
Yes. Let's refactor it later.